Sources/SwiftSyntax/SwiftcInvocation.swift

```diff
@@ -52,15 +52,36 @@ private func runCore(_ executable: URL, _ arguments: [String] = [])
     -> ProcessResult {
   let stdoutPipe = Pipe()
   var stdoutData = Data()
+
+#if os(macOS)
   stdoutPipe.fileHandleForReading.readabilityHandler = { file in
     stdoutData.append(file.availableData)
   }
+#else
+  // Temporary fix until swift-corelibs-foundation supports .readabilityHandler
+  let stdoutSource = DispatchSource.makeReadSource(
```

Comment:
Would it be possible to also use `DispatchSource` on macOS? That way we wouldn't need to differentiate between macOS and Linux which would feel desirable to me.

Reply:
I'd prefer to keep it like this for now just incase the solution for Linux is missing something as I don't want to regress the macOS part, but I will get the fix into corelibs-foundation asap so that this temp fix can be removed.

Reply:
Having it consistent between the two platforms would mean that the two hit the same issues. That way we would catch any bugs earlier when developing on Mac because it also fails on that platform.

Reply:
@ahoppen I've updated it to use `makeReadSource` on both platforms.